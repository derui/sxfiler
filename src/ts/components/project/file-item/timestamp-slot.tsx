import * as React from "react";
import { styled } from "@/components/theme";

export type Props = {
  timestamp: Date;
};

/**
 * format Date to display in item
 */
const format = function format(timestamp: Date): string {
  const year = `${timestamp.getFullYear()}`.padStart(4, "0");
  const month = `${timestamp.getMonth() + 1}`.padStart(2, "0");
  const date = `${timestamp.getDate()}`.padStart(2, "0");

  return `${year}/${month}/${date}`;
};

const Timestamp = styled.span`
  flex: 0 1 auto;
  padding: 0 ${props => props.theme.spaces.small};
  text-align: right;

  white-space: nowrap;
`;

export const Component: React.FC<Props> = prop => {
  const date = format(prop.timestamp);
  return <Timestamp>{date}</Timestamp>;
};
