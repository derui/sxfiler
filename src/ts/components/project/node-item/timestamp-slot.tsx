import * as React from "react";

interface Prop {
  timestamp: Date;
}

/**
 * format Date to display in item
 */
function format(timestamp: Date): string {
  const year = `${timestamp.getFullYear()}`.padStart(4, "0");
  const month = `${timestamp.getMonth() + 1}`.padStart(2, "0");
  const date = `${timestamp.getDate()}`.padStart(2, "0");

  return `${year}/${month}/${date}`;
}

const NodeTimestamp: React.FC<Prop> = prop => {
  const date = format(prop.timestamp);
  return <span className="fp-FileItem_Timestamp">{date}</span>;
};

export default NodeTimestamp;
