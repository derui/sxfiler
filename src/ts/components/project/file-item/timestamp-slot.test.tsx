import { h } from "preact";
import { render } from "preact-render-to-string";
import { Component as T } from "./timestamp-slot";

describe("Project", () => {
  describe("Node Item", () => {
    describe("Timestamp slot", () => {
      it("should show formatted timestamp", () => {
        const date = new Date("2019-02-28T01:02:03Z");
        const tree = render(<T timestamp={date} />);

        expect(tree).toMatchSnapshot();
      });
    });
  });
});
