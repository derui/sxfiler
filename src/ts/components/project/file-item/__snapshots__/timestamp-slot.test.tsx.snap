// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Node Item Timestamp slot should show formatted timestamp 1`] = `
<span
  className="timestamp"
>
  2019/02/28
</span>
`;
